### PR TITLE
Fixed crashes when there are NaN values in dataframe

### PR DESCRIPTION
**Pull request by GoldenCorgi**
Currently, a common way to read csv files (from kaggle, etc) to dataframe is using the below code

```python
df = pd.read_csv("example.csv")
hip.Experiment.from_dataframe(df).display()
```
which replaces empty string with NaN, which crashes when the second line is run.

A workaround used in the colab notebook was
```python
df = pd.read_csv("example.csv", keep_default_na=False)
hip.Experiment.from_dataframe(df).display()
```

I propose a change in the from_dataframe code such that it automatically replace NaN back with empty strings, so that it will be coerced into null by the from_dict code.

I can change back the colab notebook example to the first codeblock once the latest version of HiPlot rolls out with the fix!

### CODE REVIEW

**Comment by GoldenCorgi** [comment]
Not sure how the git commit PRs work yet, I apologise first if the commits are abit messy (not sure why my previous commits for previous PR are showing). Do let me know and I can try to fix up the commits

**Comment by danthe3rd** [comment]
Thank you for contributing (again today) to HiPlot! I wasn't aware of this issue, thanks for pointing out and suggesting a fix :)

Regarding the solution you propose, I would be happy to merge it once we address the following issues:
1. This will replace `NaN` in other columns as well, which is currently well supported by HiPlot
```python
import pandas as pd
df = pd.DataFrame(data={'a':[1,2,3,None],'b':[4,5,None,6],'c':[None,None,7,8]})
print(df)
hip.Experiment.from_dataframe(df).display()
```
2. I believe your solution modifies the DataFrame passed as argument (which can be confusing for the user)
I think doing something like `dataframe = dataframe.fillna({'from_uid': '', 'uid': ''})` should work :)

> Not sure how the git commit PRs work yet, I apologise first if the commits are abit messy (not sure why my previous commits for previous PR are showing). Do let me know and I can try to fix up the commits

This is because I rebased your last PR and it got merged as a [single commit](https://github.com/facebookresearch/hiplot/commit/3710caa014ddf76e9b8d09e14ef68e22723f6412). Don't worry about that, I will do the same here and the commit history won't appear :)

**Comment by danthe3rd** [comment]
Oh also it would be awesome if you could add a corresponding test case in [this file](https://github.com/facebookresearch/hiplot/blob/master/hiplot/test_experiment.py).
You can run it with `pytest ./hiplot/text_experiment.py` (you'll need to `pip install -r requirements/dev.txt` first)

**Comment by GoldenCorgi** [comment]
Hello @danthe3rd ! Thanks so much for the pointers, really do help a lot since I'm still new to this! I've added your specified changes, along with some column type checking,

```python
df = pd.DataFrame(data={'uid':[1,2,3,4],'from_uid':[None,1,2,3],'a':[1,2,3,None],'b':[4,5,None,6]})
```
creates a uid column with "int64" and a from_uid column with "float" as NaN values are inherently handled as floats by pandas
https://pandas.pydata.org/pandas-docs/stable/user_guide/integer_na.html

```python
# Check if from_uid and uid is both in columns
if {'from_uid', 'uid'}.issubset(dataframe.columns):
    # Check if there are any NaN values to handle
    if dataframe['from_uid'].isnull().values.any():

        # NaN values forces integer columns to become float, if uid is integer and from_uid is float, it crashes.
        # The line below changes uid to match from_uid type (either float or string), since NaN cannot be integer. 
        dataframe['uid']= dataframe['uid'].astype(dataframe['from_uid'].dtypes)
        dataframe = dataframe.fillna({'from_uid': '', 'uid': ''})

        # Replaces their dtypes accordingly to str, which is handled better with lesser errors with no change to functionality
        dataframe['uid']= dataframe['uid'].astype(str)
        dataframe['from_uid']= dataframe['from_uid'].astype(str)
```

I've also added two test cases, one for string - None, and floats - NaN!

There is still a crashing issue where from_uid exists in the dataframe, but uid does not, 
```python
df = pd.DataFrame(data={'from_uid':[None,1,2,3],'a':[1,2,3,None],'b':[4,5,None,6]})
```
but I believe this is a dataset/user issue (uid should always exist when from_uid exist), therefore I did not specify any fixes for this. 

Do let me know if this is an issue too, and I can try to see if I can help to raise an exception when this scenario occurs!

File: requirements/dev.txt [hunk 1]
```diff
@@ -10,3 +10,4 @@ guzzle_sphinx_theme
 pre-commit
 pandas
 streamlit>=0.63
+beautifulsoup4
```
Comment (danthe3rd) [inline requirements/dev.txt:13]: nice catch thanks!

**Comment by danthe3rd** [comment]
Thank you for your contribution :)

> There is still a crashing issue where from_uid exists in the dataframe, but uid does not,
> 
> ```python
> df = pd.DataFrame(data={'from_uid':[None,1,2,3],'a':[1,2,3,None],'b':[4,5,None,6]})
> ```
> 
> but I believe this is a dataset/user issue (uid should always exist when from_uid exist), therefore I did not specify any fixes for this.

Yes exactly, that's a user issue. Indeed we could be more user friendly, but that's beyond the scope of this change
